### DISCUSSION TITLE
A question about particle mover (Particle.F90)

### DISCUSSION THREAD

**Discussion by FemtoSimin** [epochpic/epoch#343] [Q&A] [answered]
If c of dtco2 is the light speed in vacuum,  is part_ux the particle speed in unit of c? Or is there more treatment to part_x involving c regarding the calculation of current J? Otherwise, the line "delta_x = part_ux * root" has an extra c compared to Eq. 9 in Arber, T. D., et al. Plasma Physics and Controlled Fusion 57.11 (2015): 113001. I am confused here, will appreciate any answer!

Code 1:
    dto2 = dt / 2.0_num
    dtco2 = c * dto2

Code 2:
  ! Calculate particle velocity from particle momentum
        part_u2 = part_ux**2 + part_uy**2 + part_uz**2
        gamma_rel = SQRT(part_u2 + 1.0_num)
        root = dtco2 / gamma_rel

        delta_x = part_ux * root
        delta_y = part_uy * root
        delta_z = part_uz * root

        ! Move particles to end of time step at 2nd order accuracy
        part_x = part_x + delta_x
        part_y = part_y + delta_y
        part_z = part_z + delta_z

**Comment by TomGoffrey** [comment, marked as answer]
Yes, `part_ux`, `part_uy` and `part_uz` are normalised by c. See also the calculation of gamma just above the calculation of `delta_x`.

**Reply by FemtoSimin** [reply]
Thank you!
